ethsigner/signer/multifile-based/src/main/java/tech/pegasys/ethsigner/signer/multifilebased/KeyPasswordLoader.java

```diff
@@ -42,31 +42,43 @@
   }
 
   Optional<KeyPasswordFile> loadKeyAndPasswordForAddress(final String address) {
-    final String filename = normalizeAddress(address);
-    final Path keyPath = keysDirectory.resolve(filename + KEY_FILE_EXTENSION);
-    if (keyPath.toFile().exists() && isFilenameValid(keyPath)) {
-      return tryFindingMatchingPassword(keyPath).map(path -> new KeyPasswordFile(keyPath, path));
-    } else {
+    final List<KeyPasswordFile> matchingKeys =
+        loadAvailableKeys().stream()
+            .filter(kp -> kp.getName().toLowerCase().contains(normalizeAddress(address)))
+            .collect(Collectors.toList());
+
+    if (matchingKeys.size() > 1) {
+      LOG.error("Found multiple key/password matches for address {}", address);
       return Optional.empty();
+    } else if (matchingKeys.isEmpty()) {
+      return Optional.empty();
+    } else {
+      return Optional.of(matchingKeys.get(0));
     }
   }
 
-  Collection<KeyPasswordFile> loadAvailableKeys() throws IOException {
+  Collection<KeyPasswordFile> loadAvailableKeys() {
     final Collection<KeyPasswordFile> keysAndPasswords = new HashSet<>();
 
     try (final DirectoryStream<Path> directoryStream = Files.newDirectoryStream(keysDirectory)) {
-      for (final Path keyFile : directoryStream) {
-        if (KEY_FILE_MATCHER.matches(keyFile) && isFilenameValid(keyFile)) {
-          tryFindingMatchingPassword(keyFile)
+      for (final Path file : directoryStream) {
+        if (isKeyFile(file)) {
```

Comment:
Instead of doing an if here it would be nicer to use a glob pattern or filter with the directory stream